### ISSUE TITLE
tab_width should fall back to indent_size

### ISSUE THREAD

**Issue by sharwell** [dotnet/roslyn#22601]
**Version Used**: 15.1

:link: Originally posted on [developer community](https://developercommunity.visualstudio.com/content/problem/46983/conflict-between-vs2017-text-editor-indentation-se.html) by Clinton Ingram (@saucecontrol?).

**Steps to Reproduce**:

Use the following **.editorconfig** file:

```
[*.cs]
indent_style = tab
indent_size = 16
```

**Expected Behavior**:

The editor formats code with a single hard-tab for indentation, which is displayed in the editor with a width of 16 characters. According to the EditorConfig format:

> `tab_width`: a whole number defining the number of columns used to represent a tab character. This defaults to the value of `indent_size` and doesn't usually need to be specified.

**Actual Behavior**:

The editor formats code with multiple hard tabs for indentation.

**Impacted code**:

https://github.com/dotnet/roslyn/blob/a703709e013d6f7a135f6097f647ace993ee0ffd/src/Workspaces/Core/Portable/Formatting/FormattingOptions.cs#L14-L17

**Comment by sharwell** [comment]
:memo: We also have an internal bug tracking editor-related functionality impacted by this interpretation of the EditorConfig file.

**Comment by MhdTlb** [comment]
Can I work on this?

**Comment by sharwell** [comment]
@MhdTlb Yes. Do keep in mind that I'm not certain this is an issue that may or may not have an easy solution. If you get stuck, let us know and we'll try to track down the right person to unblock you. :smile:

**Comment by Gargaj** [comment]
This has a weird effect on the auto-formatter (Format Selection or Format Document) in that if there's a difference between `tab_width` and `indent_size`, it will use the incorrect amount of indentations.

**Comment by Eilon** [comment]
Got here from https://github.com/dotnet/format/issues/574 because this affects running `dotnet format` where you get a weird mix of tabs and spaces in the same document. Fortunately, the workaround of setting `tab_width` to the same value as `indent_size` works.
